docs/advanced_usages/scaling.md

```diff
@@ -0,0 +1,635 @@
+    ```
+
+    Change `ANTA_INVENTORY` and `TEST_CATALOGS_PATH` to match your environment.
+
+2. Create the parallel execution script (`run.sh`):
+
+    ```bash
+    #!/bin/bash
+
+    mkdir -p nrfu_reports
+
+    NODE_NAMES=$(yq '.anta_inventory.hosts[].name' < "anta_inventory.yml" | tr -d '"')
+
+    parallel --tag './nrfu.sh {} nrfu_reports' ::: $NODE_NAMES || NRFU_STATUS=$?
+
+    exit $NRFU_STATUS
+    ```
+
+    Also change `anta_inventory.yml` to match your inventory file specified in `ANTA_INVENTORY` of `nrfu.sh`.
+
+3. Make both scripts executable:
+
+    ```bash
+    chmod +x nrfu.sh run.sh
+    ```
+
+4. Run the script:
+
+    ```bash
+    ./run.sh
+    ```
+
+5. The script will execute ANTA tests for each device in parallel, generating a JSON report for each device in the `nrfu_reports` directory.
+
+#### Tag-Based Implementation
+
+1. Create the tag-specific test runner (`nrfu-tag.sh`):
+
+    ```bash
+    #!/bin/bash
+    set -euo pipefail
+
+    exec 2>&1
+
+    usage() {
+        echo "Usage: $0 <tag-name> [<nrfu-reports-path>]"
+        exit 1
+    }
+
+    ANTA_INVENTORY="anta_inventory.yml"
+    ANTA_CATALOG="anta_catalog.yml"
+    TAG_NAME=${1:-}
+    NRFU_REPORT_PATH=${2:-nrfu_reports}
+
+    if [[ -z "$TAG_NAME" ]]; then
+        usage
+    fi
+
+    anta nrfu \
+        --tags "$TAG_NAME" \
+        -i "$ANTA_INVENTORY" \
+        -c "$ANTA_CATALOG" \
+        json -o "$NRFU_REPORT_PATH/${TAG_NAME}.json"
+    ```
+
+    Same as before, adjust `ANTA_INVENTORY`, `ANTA_CATALOG` to match your environment.
+
+2. Create the parallel execution script (`run-tag.sh`):
+
+    ```bash
+    #!/bin/bash
+
+    mkdir -p nrfu_reports
+
+    TAGS=$(yq -r '.. | select(.filters?) | .filters.tags[]' < "anta_catalog.yml" | sort -u)
+
+    parallel --tag './nrfu_tag.sh {} nrfu_reports' ::: $TAGS || NRFU_STATUS=$?
+
+    exit $NRFU_STATUS
+    ```
+
+    Change `anta_catalog.yml` to match your catalog file specified in `ANTA_CATALOG` of `nrfu-tag.sh`.
+
+3. Make both scripts executable:
+
+    ```bash
+    chmod +x nrfu-tag.sh run-tag.sh
+    ```
+
+4. Run the script:
+
+    ```bash
+    ./run-tag.sh
+    ```
+
+5. The script will execute ANTA tests for each tag in parallel, generating a JSON report for each tag in the `nrfu_reports` directory.
+
+### Performance Considerations
+
+#### Parallel Job Control
+
+The `-j` option in `GNU Parallel` controls the number of concurrent jobs. Consider these factors when setting the job limit:
+
+1. **Available CPU cores**:
+
+     - Recommend: jobs = number of cores - 1
+     - Example: For 8 cores, use `-j7`
+
+2. **Memory requirements**:
+
+     - Each ANTA instance requires memory
+     - Monitor system resources during execution using `htop` or similar tools
+
+#### JSON vs YAML
+
+When working with large catalogs and inventories, using JSON format instead of YAML can provide better performance as ANTA loads JSON files more efficiently. If you choose to use JSON:
```

Comment:
I think thats fair?
```suggestion
When working with large catalogs and inventories, using JSON format instead of YAML can provide better performance as Python loads JSON files more efficiently. If you choose to use JSON:
```